test/e2e/cmd/rabbitproducer/main.go

```diff
@@ -68,17 +68,47 @@ func main() {
 		nil,      // arguments
 	)
 	failOnError(err, "Failed to declare an exchange")
+	var body, contentType string
+	var headers amqp.Table
 
 	for i := 0; i < env.Count; i++ {
-		body := fmt.Sprintf(`{ "id": %d, "message": "Hello, World!" }`, i)
+		if i < env.Count/3 {
+			headers = amqp.Table{
+				"ce-specversion":     "1.0",
+				"ce-id":              i,
+				"ce-type":            "knative.producer.e2etest",
+				"ce-source":          "example/source.uri",
+				"ce-datacontenttype": "application/json; charset=UTF-8",
+			}
+			body = `{ "message": "Hello, BinCEWorld!" }`
+		} else if i < (env.Count/3)*2 {
+			contentType = "application/cloudevents+json"
+			headers = amqp.Table{}
+			body = fmt.Sprintf(`{
+				"id": %d,
+				"type": "knative.producer.e2etest",
+				"source": "example/source.uri",
+				"data": "Hello, CEWorld!",
+				"specversion": "1.0",
+				"datacontenttype": "text/plain"
+			}`,
+				i,
+			)
+		} else {
+			contentType = "text/plain"
+			headers = amqp.Table{}
+			body = fmt.Sprintf(`{ "id": %d, "message": "Hello, World!" }`, i)
+		}
```

Comment:
This way seems more straightforward, alternating between all three. Also adds content-type for structured mode.

We probably want a test that asserts how many binary/structured/wrapped messages we get out, but that's ok to not include here.
```suggestion
		switch i % 3 {
		case 0:
			contentType = "application/json"
			headers = amqp.Table{
				"ce-specversion":     "1.0",
				"ce-id":              i,
				"ce-type":            "knative.producer.e2etest",
				"ce-source":          "example/source.uri",
				"ce-datacontenttype": "application/json; charset=UTF-8",
			}
			body = `{ "message": "Hello, BinCEWorld!" }`
		case 1:
			contentType = "application/cloudevents+json"
			headers = amqp.Table{}
			body = fmt.Sprintf(`{
				"id": %d,
				"type": "knative.producer.e2etest",
				"source": "example/source.uri",
				"data": "Hello, CEWorld!",
				"specversion": "1.0",
				"datacontenttype": "text/plain"
			}`, i)
		case 2:
			contentType = "text/plain"
			headers = amqp.Table{}
			body = fmt.Sprintf(`{ "id": %d, "message": "Hello, World!" }`, i)
		}
```

Reply:
kk i like the switch case